src/main/java/at/porscheinformatik/seleniumcomponents/clarity/ClarityInputContainerComponent.java

```diff
@@ -70,4 +72,14 @@ public void clear()
         // Mark the whole input text and delete it
         input.sendKeys(Keys.chord(Keys.CONTROL, "A"), Keys.BACK_SPACE);
     }
+
+    /**
+     * Use this method to check whether or not you can write a new value to the input control.
+     * @return true if the input control has been disabled, and true otherwise
+     */
+    public boolean isDisabled()
```

Comment:
I think this check should take place when writing a new value

```java
public void setValue(String value)
{
    // or something similar
    if (!isDisabled())
    {
        input.enter(value);
    }
}
```

Reply:
I would not implement this check. Checking for `isDisabled` needs at least one request to the Browser. When running in the cloud via Browserstack or some other tool, this request is relatively slow.

I would guess, that 90% of the input fields in a form are not disabled. So performing this check for every input field, will slow the tests down for no reason.
Normally you know, when a Input Field is disabled or not, and do not even set a value in this case.

And another disadvantage of this check is, that it now silently ignores the value. It is not easy to spot failures in the Test anymore. Without the check, it throws an exception at exactly this line, and you know that you tried to set a value on a disabled element.